service/lib/dinstaller/manager.rb

```diff
@@ -133,7 +133,11 @@ def install
     #
     # @return [Software]
     def software
-      @software ||= DBus::Clients::Software.new
+      return @software if @software
+
+      @software = DBus::Clients::Software.new
+      @software.on_product_selected { probe }
```

Comment:
Cool! I think the *software* command should configure this callback too.

Reply:
Something like:

~~~ruby
software_client.on_product_selected { manager_client.probe } 
~~~

Reply:
I prefer to just raise a signal instead of calling back the manager, so the software component is as decoupled as possible from the manager service.

Reply:
You are totally right! The signal is already emitted and the manager service reacts to it. Thanks!